pallets/ocex/src/lib.rs

```diff
@@ -901,98 +928,108 @@ pub mod pallet {
 			// This is to build services that can enable free withdrawals similar to CEXes.
 			let _ = ensure_signed(origin)?;
 			// This vector will keep track of withdrawals processed already
-			let mut processed_withdrawals: BoundedVec<Withdrawal<T::AccountId>, WithdrawalLimit> =
-				BoundedVec::<Withdrawal<T::AccountId>, WithdrawalLimit>::default();
+			let mut processed_withdrawals = vec![];
+			let mut failed_withdrawals = vec![];
 			ensure!(
 				<Withdrawals<T>>::contains_key(snapshot_id),
 				Error::<T>::InvalidWithdrawalIndex
 			);
 			// This entire block of code is put inside ensure as some of the nested functions will
 			// return Err
-			ensure!(
-				<Withdrawals<T>>::mutate(snapshot_id, |btree_map| {
-					// Get mutable reference to the withdrawals vector
-					if let Some(withdrawal_vector) = btree_map.get_mut(&account) {
-						while withdrawal_vector.len() > 0 {
-							// Perform pop operation to ensure we do not leave any withdrawal left
-							// for a double spend
-							if let Some(withdrawal) = withdrawal_vector.pop() {
-								if let Some(converted_withdrawal) = withdrawal
-									.amount
-									.saturating_mul(Decimal::from(UNIT_BALANCE))
-									.to_u128()
+			<Withdrawals<T>>::mutate(snapshot_id, |btree_map| {
+				// Get mutable reference to the withdrawals vector
+				if let Some(withdrawal_vector) = btree_map.get_mut(&account) {
+					while withdrawal_vector.len() > 0 {
+						// Perform pop operation to ensure we do not leave any withdrawal left
+						// for a double spend
+						if let Some(withdrawal) = withdrawal_vector.pop() {
+							if let Some(converted_withdrawal) = withdrawal
+								.amount
+								.saturating_mul(Decimal::from(UNIT_BALANCE))
+								.to_u128()
+							{
+								if Self::transfer_asset(
+									&Self::get_pallet_account(),
+									&withdrawal.main_account,
+									converted_withdrawal.saturated_into(),
+									withdrawal.asset,
+								)
+								.is_ok()
 								{
-									if Self::transfer_asset(
-										&Self::get_pallet_account(),
-										&withdrawal.main_account,
-										converted_withdrawal.saturated_into(),
-										withdrawal.asset,
-									)
-									.is_ok()
-									{
-										processed_withdrawals
-											.try_push(withdrawal)
-											.unwrap_or_default();
-									} else {
-										// Storing the failed withdrawals back into the storage item
-										withdrawal_vector
-											.try_push(withdrawal.clone())
-											.unwrap_or_default();
-										Self::deposit_event(Event::WithdrawalFailed(withdrawal));
-									}
+									processed_withdrawals.push(withdrawal.to_owned());
```

Comment:
Same

Reply:
not sure what's the problem here... elaborate, please

Reply:
Ye, sorry, related to pushes and panics like [here](https://github.com/Polkadex-Substrate/Polkadex/pull/579#discussion_r1001172687).

Reply:
that's regular vector now - no panics on push